### PR TITLE
docs: remove multiport docs, add v1dns flag

### PR DESCRIPTION
**Pull request by DanStough**
### Description
This PR is a docs update for:
1. Removing multiport docs since we no longer plan to support this feature.
2. Document the `v1dns` feature flag, which would only be needed to rollback the refactored DNS server.

### CODE REVIEW

File: website/content/docs/k8s/helm.mdx [hunk 1]
```diff
@@ -763,25 +763,14 @@ Use these links to navigate to a particular top-level stanza.
...
-      _**Warning**_! This feature is under active development. It is not
-      recommended for production use. Setting this flag during an
-      upgrade could risk breaking your Consul cluster.
-      If this flag is set, Consul V2 resources (catalog, mesh, auth, etc)
-      will use V2 implementations for tenancy (partitions and namesapces)
-      instead of bridging to the existing V1 implementations. The
-      `resource-apis` feature flag must also be set.
+    - `v1dns`:
+      If this flag is set, Consul agents will use the legacy DNS implementation.
+      This would only be required in the case of a DNS bug found in the refactoring introduced in 1.19.
```
Comment (DanStough) [inline website/content/docs/k8s/helm.mdx:768]: I'm realizing now that I will also need to backport this fix to the k8s `values.yaml`.

File: website/content/docs/release-notes/consul/v1_17_x.mdx [hunk 2]
```diff
@@ -15,14 +15,10 @@ We are pleased to announce the following Consul updates.
 
   These APIs are the foundation for future versions of Consul and enable new functionalities, such as multi-port and host-name-based canary routing and routing traffic through headless services in native Kubernetes deployments.
 
-  For more information, refer to the [Catalog API v2](/consul/docs/k8s/multiport#catalog-api-v2-beta) section in the documentation.
-
   <Note> These APIs are in beta and under active development, so we do not recommend using them in production. </Note>
 
 - **Multi-port services in Consul:** You can now register services with multiple ports per service. The v2 catalog API enables a single sidecar proxy to support workloads on different ports. This significantly reduces the operational overhead for managing Consul service mesh. Support for other runtimes outside of Kubernetes is planned for future releases of Consul.
 
-  Refer to the [Multi-port services for service mesh](/consul/docs/k8s/multiport#catalog-api-v2-beta) and [Configure multi-port services](/consul/docs/k8s/multiport/configure) for more information.
```
Comment (zalimeni) [inline website/content/docs/release-notes/consul/v1_17_x.mdx:24]: ~ Given these are part of release notes, we could instead link to the versioned copies, assuming we don't intend to backport these deletions to versioned docs:
https://developer.hashicorp.com/consul/docs/v1.17.x/k8s/multiport
https://developer.hashicorp.com/consul/docs/v1.17.x/k8s/multiport/configure

**Review by zalimeni — APPROVED** [review]
LGTM - thank you for combing through and finding these.

File: website/content/docs/release-notes/consul/v1_17_x.mdx (hunk 2 above)
Comment (DanStough) [reply website/content/docs/release-notes/consul/v1_17_x.mdx:24]: Ahh, that's a much better idea! My first attempt was pure laziness.

File: website/content/docs/k8s/helm.mdx [hunk 3]
```diff
@@ -763,25 +763,14 @@ Use these links to navigate to a particular top-level stanza.
...
 
     Example:
 
     ```yaml
-    experiments: [ "resource-apis" ]
+    experiments: [ "v1dns" ]
     ```
```
[the remark this answers is not in the document]
Comment (DanStough) [reply website/content/docs/k8s/helm.mdx:775]: Yep, I have a PR to backport these changes here: https://github.com/hashicorp/consul-k8s/pull/4092

File: website/content/docs/k8s/helm.mdx (hunk 1 above)
[the remark this answers is not in the document]
Comment (DanStough) [reply website/content/docs/k8s/helm.mdx:768]: I'll update my other PR with this edit. 👍

File: website/content/docs/architecture/catalog.mdx [hunk 4]
```diff
@@ -14,7 +14,9 @@ For more information about the information returned when querying the catalog, i
 
 Consul tracks information about registered services through its catalog API. This API records user-defined information about the external services, such as their partitions and required health checks. It also records information that Consul assigns for its own operations, such as an ID for each service instance and the [Raft indices](/consul/docs/architecture/consensus) when the instance is registered and modified.
 
-Consul uses v1 of the catalog API by default. Consul v1.17.0 and later ships with version 2 (v2) of the catalog API. V2 is intended for testing and development purposes. V1 and V2 of the catalog APIs cannot run concurrently in a Consul deployment. There is no migration path between catalog versions. For more information, refer to [Consul v2 Catalog API](/consul/docs/architecture/v2/catalog).
+### v2 Catalog
+
+Consul introduced an experimental v2 Catalog API in v1.17.0. This API supported multi-port Service configurations on Kubernetes, and it was made available for testing and development purposes. The v2 catalog and its support for multiport Kubernetes Services were deprecated in the v1.19.0 release.
```
Comment (DanStough) [inline website/content/docs/architecture/catalog.mdx:19]: 👍

**Pull request merged**